sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -390,6 +425,15 @@ def _collect_plans(self, rows, cursor, deadline):
                 self.check.count("dd.sqlserver.statements.deadline_exceeded", 1, **self.check.debug_stats_kwargs())
                 return
             plan_key = (row['query_signature'], row['query_hash'], row['query_plan_hash'])
+            # for stored procedures, we only want to look up plans for the entire procedure
+            # not every query that is executed within the proc. Queries from the same procedure
+            # have the same plan_handle, but different query signatures & can have diff plan hashes.
+            # taking the first 16 bytes should be unique enough to properly identify the plan.
+
+            # TODO: technically this could be the proc sig and the plan handle?? I believe that would
```

Comment:
Yes, now that we have the procedure signature it's probably better to switch to using that as the rate limiting key.

Reply:
ok, cool 👍